frontend/src/metabase/admin/settings/components/EmbeddingSettings/hooks.tsx

```diff
@@ -0,0 +1,27 @@
+import { useMemo } from "react";
+
+import { useSelector } from "metabase/lib/redux";
+import type { SettingKey, Settings } from "metabase-types/api";
+
+type DisplaySetting<Key extends SettingKey> = { key: Key };
+type SettingWithValue<Key extends SettingKey> = {
+  key: Key;
+  value: Settings[Key];
+  [key: string]: any;
+};
+
+export function useMergeSetting<Key extends SettingKey>(
+  displaySetting: DisplaySetting<Key>,
+): SettingWithValue<Key> {
+  const apiSetting = useSelector(state => state.admin.settings.settings).find(
+    (setting: any) => setting.key === displaySetting.key,
+  ) as unknown as SettingWithValue<Key>;
```

Comment:
couldn't figure out a better way than this yet. I make it so that the return value works by having a proper `key`.